spring-cloud-deployer-spi/src/test/java/org/springframework/cloud/deployer/spi/app/observation/ObservedAppDeployerTests.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.app.observation;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.tck.MeterRegistryAssert;
+import io.micrometer.tracing.Span;
+import io.micrometer.tracing.test.SampleTestRunner;
+import io.micrometer.tracing.test.simple.SpansAssert;
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.deployer.spi.app.AppDeployer;
+import org.springframework.cloud.deployer.spi.app.AppScaleRequest;
+import org.springframework.cloud.deployer.spi.app.AppStatus;
+import org.springframework.cloud.deployer.spi.core.AppDefinition;
+import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
+import org.springframework.cloud.deployer.spi.core.RuntimeEnvironmentInfo;
+import org.springframework.core.io.PathResource;
+
+/**
+ * Tests for {@link AppDeployer}
+ */
+public class ObservedAppDeployerTests extends SampleTestRunner {
+
+	ObservedAppDeployerTests() {
+		super(SampleRunnerConfig.builder().build());
+	}
+
+	@Override
+	public SampleTestRunnerConsumer yourCode() {
+		return (bb, meterRegistry) -> {
+			AppDeployer appDeployer = new ObservedAppDeployer(getObservationRegistry(), appDeployer(), 100L);
+
+			String id = appDeployer.deploy(deploymentRequest());
+			appDeployer.status(id);
+			appDeployer.scale(new AppScaleRequest(id, 2));
+			appDeployer.undeploy(id);
+
+			SpansAssert.assertThat(bb.getFinishedSpans())
+					.haveSameTraceId()
+					.hasASpanWithName("deploy", spanAssert -> spanAssert.hasRemoteServiceNameEqualTo("Test")
+							.hasTag("spring.cloud.deployer.app.id", "Deployment request received")
```

Comment:
Is this what the app id should be?

Reply:
That's what I put in here https://github.com/spring-cloud/spring-cloud-deployer/pull/360/files/f0b16ccfb96ca1e1e68be28f163944fa931b206a#diff-bc19637713d0dd0728458d6c4b297b715590a3d59243646e727e700d02efd16bR106 :shrug: